plugins/extractors/http/execute_script.go

```diff
@@ -0,0 +1,169 @@
+package http
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/d5/tengo/v2"
+	"github.com/odpf/meteor/models"
+	v1beta2 "github.com/odpf/meteor/models/odpf/assets/v1beta2"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/plugins/internal/tengoutil"
+	"github.com/odpf/meteor/plugins/internal/tengoutil/structmap"
+	"google.golang.org/protobuf/proto"
+)
+
+func (e *Extractor) executeScript(ctx context.Context, res interface{}, emit plugins.Emit) error {
+	s := tengoutil.NewSecureScript(([]byte)(e.config.Script.Source))
+	if err := declareGlobals(s, res, emit); err != nil {
+		return fmt.Errorf("declare globals: %w", err)
+	}
+
+	c, err := s.Compile()
+	if err != nil {
+		return fmt.Errorf("compile: %w", err)
+	}
+
+	if err := c.RunContext(ctx); err != nil && !errors.Is(err, errUserExit) {
+		return fmt.Errorf("run: %w", err)
+	}
+
+	return nil
+}
+
+func declareGlobals(s *tengo.Script, res interface{}, emit plugins.Emit) error {
+	for name, v := range map[string]interface{}{
+		"response": res,
+		"new_asset": &tengo.UserFunction{
+			Name:  "new_asset",
+			Value: newAssetWrapper(),
+		},
+		"emit": &tengo.UserFunction{
```

Comment:
is it possible to hide the emit context internally and when user creates the script, it would be like returning only?

Reply:
Modifications done in the script do not reflect in the value in Go space. We can pull out values from the script as long as it was declared in the global scope. But when assets are declared and populated from inside a loop, it won't be possible to automatically emit it from Go.